src/broadcast/setup.ts

```diff
@@ -30,14 +33,57 @@ export default function setupBroadcastIpc({
   let spectateWorker: SpectateWorker | undefined;
   let broadcastWorker: BroadcastWorker | undefined;
   let prefixOrdinal = 0;
+  let spectateIdleTimer: NodeJS.Timeout | undefined;
+
+  // Helper to start/reset the idle timeout (only when no active broadcasts)
+  const resetSpectateIdleTimeout = async () => {
+    // Clear any existing timer
+    if (spectateIdleTimer) {
+      clearTimeout(spectateIdleTimer);
+      spectateIdleTimer = undefined;
+    }
+
+    // Only start timeout if worker exists and has no active broadcasts
+    if (spectateWorker) {
+      const openBroadcasts = await spectateWorker.getOpenBroadcasts().catch(() => []);
```

Comment:
if my understanding of the event loop is correct, this should be currently safe from race conditions ([after requested changes] reading and mutating the underlying `openBroadcasts` `Record` only happens in synchronous functions), but a future change to make `SpectateManager` `watchBroadcast` async could unknowingly introduce that risk.

therefore I think it would be prudent to add a lock betwteen `SpectateManager` `watchBroadcast` and `getOpenBroadcasts`